Add Ariadne GraphQL error integration

sentrivana opened the pull request:
Capture GraphQL errors when using Ariadne server side and add more context to them (request, response).

One of three three integrations for https://github.com/getsentry/sentry-python/issues/2257

antonpirker commented on sentry_sdk/integrations/ariadne.py:
```diff
@@ -0,0 +1,175 @@
+                        error,
+                        client_options=hub.client.options,
+                        mechanism={
+                            "type": hub.get_integration(AriadneIntegration).identifier,
```
you could use the `integration` from line 70

antonpirker commented on sentry_sdk/integrations/ariadne.py:
```diff
@@ -0,0 +1,175 @@
+                        error,
+                        client_options=hub.client.options,
+                        mechanism={
+                            "type": hub.get_integration(AriadneIntegration).identifier,
```
you could use the integration from line 98

antonpirker commented on sentry_sdk/integrations/ariadne.py:
```diff
@@ -0,0 +1,175 @@
+        if integration is None:
+            return old_handle_query_result(result, *args, **kwargs)
+
+        response = old_handle_query_result(result, *args, **kwargs)
```
Nit: in the other functions it is called `result` and here `response`. if on purpose ok.

sentrivana replied:
The first arg of the function is called `result`, so can't use that here, changed `response` to `query_result` at least.

antonpirker commented on sentry_sdk/integrations/ariadne.py:
```diff
@@ -0,0 +1,175 @@
+                request_info["data"] = data
+
+            elif event.get("request", {}).get("data"):
+                del event["request"]["data"]
```
is this set somewhere outside of the integrations? and if yes, are we overwriting it in line 150?

sentrivana replied:
The event processors from our web framework integrations add this (tested at least with Flask and FastAPI). So line 150 would be overwriting stuff if something was already there. I'll make it conditional, i.e., only set `data` if not set (so that you also get the request data if you don't have a suitable web framework integration enabled).

antonpirker reviewed (approved):
Some nitpicking, otherwise: really nice clean integration!

sentrivana commented:
@antonpirker Made two additional changes since your review:
* added an `ignore_logger` to prevent the `logging` integration from capturing the error first: https://github.com/getsentry/sentry-python/pull/2387/commits/25495d565b8d4b12733d4f0cbf0a33233b0f6a28
* changed attaching `request.data` back so that the `ariadne` integration rewrites whatever is set there by the web framework integrations: https://github.com/getsentry/sentry-python/pull/2387/commits/2ceb9b106d5c738277647825cedeee0fcb708112 As I was testing stuff for docs I noticed that the web framework integrations sometimes seem to attach [a wrong request body](https://sentry-sdks.sentry.io/issues/4510338750/events/0630897d4bc9432bb6f179a880943736/). Not sure exactly why. Making the `ariadne` integration overwrite it ensures that it's [the correct one](https://sentry-sdks.sentry.io/issues/4510338750/events/2b6201e19eb54f5285873eff3f54c7bf/).
